lib/1.4/dml-builtins.dml

```diff
@@ -54,6 +54,14 @@ extern typedef _callback_entry_t * vect _callback_vect_t;
 extern typedef _connection_entry_t * vect _connection_vect_t;
 
 extern typedef struct {} _id_info_t;
+extern typedef struct {
+    uint32 id;
+    uint32 encoded_index;
+} _identity_t;
+extern typedef struct {
+    void *trait;
+    _identity_t id;
+} _traitref_t;
 
```

Comment:
can we hide these refs and send &this to dmllib in a void* arg instead? or at least make the traitref_t type opaque?

Reply:
For this PR? Sure. For attribute softcoding? _No_. With that in mind I don't see the point.

Reply:
Maybe make it an opaque type with accessor functions for the stuff attr softcoding needs? (I haven't reviewed that one yet so I don't really know what I'm talking about)

Reply:
attr softcoding needs construction functions -- a way to create an `attribute` out of `id`, `encoded_index` and `trait`. If you have accessor functions for that, why bother making the type opaque.

Actually, in general, why bother making the type opaque. Don't quote Hyrum's at me, how would accessor functions be any better?